client/components/plans-redesign/plan-actions/index.jsx

```diff
@@ -0,0 +1,285 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import { cartItems } from 'lib/cart-values';
+import config from 'config';
+import { isBusiness, isEnterprise, isFreePlan, isFreeJetpackPlan } from 'lib/products-values';
+import purchasesPaths from 'me/purchases/paths';
+import { isValidFeatureKey } from 'lib/plans';
+import * as upgradesActions from 'lib/upgrades/actions';
+
+const PlanActions = React.createClass( {
+	propTypes: {
+		plan: React.PropTypes.object,
+		selectedFeature: React.PropTypes.string
+	},
+
+	getButtons() {
+		if ( this.props.isImageButton ) {
+			return this.getImageButton();
+		}
+
+		if ( this.props.isInSignup ) {
+			return this.upgradeActions();
+		}
+
+		if ( ! this.props.sitePlan ) {
+			return null;
+		}
+
+		if ( this.props.site && ! this.props.site.isUpgradeable() ) {
+			return null;
+		}
+
+		if ( this.siteHasThisPlan() ) {
+			if ( this.props.sitePlan.freeTrial ) {
+				return this.upgradeActions();
+			}
+
+			return (
+				<div className="plan-actions__action-details">
+					<div className="plan-actions__current">
+						{ this.managePlanButton() }
+						{ this.getCurrentPlanHint() }
+					</div>
+				</div>
+			);
+		}
+
+		if ( this.canSelectPlan( this.props.plan ) ) {
+			return this.getInternalButtons();
+		}
+	},
+
+	getInternalButtons() {
+		if ( ! this.props.cart.hasLoadedFromServer || ! this.props.site ) {
+			return null;
+		}
+
+		return this.upgradeActions();
+	},
+
+	freePlanButton() {
+		return (
+			<div>
+				<button className="button is-primary plan-actions__upgrade-button"
+					disabled={ this.props.isSubmitting }
+					onClick={ this.handleSelectPlan }>
+					{ this.translate( 'Select Free Plan' ) }
+				</button>
+			</div>
+		);
+	},
+
+	freeJetpackPlanButton() {
+		return (
+			<div>
+				<button className="button is-primary plan-actions__upgrade-button"
+					disabled={ this.props.isSubmitting }
+					onClick={ this.handleSelectJetpackFreePlan }>
+					{ this.translate( 'Select Free Plan' ) }
+				</button>
+			</div>
+		);
+	},
+
+	upgradeActions() {
+		if ( isFreePlan( this.props.plan ) ) {
+			return this.freePlanButton();
+		}
+
+		if ( isFreeJetpackPlan( this.props.plan ) ) {
+			return this.freeJetpackPlanButton();
+		}
+
+		if ( ! config.isEnabled( 'upgrades/checkout' ) ) {
+			return null;
+		}
+
+		let label = this.translate( 'Upgrade Now' );
```

Comment:
This is just a personal preference, but if the label only changes for one condition I'd consider doing a ternary within the render return statement itself:

```
<Button>
    { condition ? labelA : labelB }
</Button>
```